Framework/include/QualityControl/Check.h

```diff
@@ -0,0 +1,87 @@
+
+#ifndef QC_CHECKER_CHECK_H
+#define QC_CHECKER_CHECK_H
+
+// std
+#include <functional>
+#include <map>
+#include <vector>
+#include <memory>
+// O2
+#include <Framework/DataProcessorSpec.h>
+// QC
+#include "QualityControl/Quality.h"
+#include "QualityControl/QualityObject.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/CheckInterface.h"
+#include "QualityControl/QcInfoLogger.h"
+
+namespace o2::quality_control::checker
+{
+
+class Check 
+{
+  public:
+    Check(std::string checkName, std::string configurationSource);
+
+    /**
+     * Expected to run in the init phase of the FairDevice
+     */
+    void init();
+    
+    std::shared_ptr<o2::quality_control::core::QualityObject> check(std::map<std::string, std::shared_ptr<o2::quality_control::core::MonitorObject>>& moMap);
+
+    // Policy 
+    /**
+     * Change the revision.
+     * Expected to be changed after invoke of `check(moMap)` or revision number overflow.
+     */
+    void updateRevision(unsigned int revision);
+    
+    /**
+     * Return true if the Monitor Objects were changed accordingly to the policy
+     */
+    bool isReady(std::map<std::string, unsigned int>& revisionMap);
+
+    const std::string getName() { return mName; };
+    std::shared_ptr<o2::quality_control::core::QualityObject> getQualityObject() { return mQualityObject; };
+    o2::framework::OutputSpec getOutputSpec() { return mOutputSpec; };
+    o2::framework::Inputs getInputs() { return mInputs; };
+
+    //TODO: Unique Input string
+    static o2::header::DataDescription createCheckerDataDescription(const std::string taskName);
+  private:
+    void initConfig();
+    void initPolicy();
+    void loadLibrary();
+    inline void beautify(std::map<std::string, std::shared_ptr<MonitorObject>>& moMap);
+
+    const std::string mName;
+    std::string mConfigurationSource;
+
+    o2::quality_control::core::QcInfoLogger& mLogger;
+    std::shared_ptr<o2::quality_control::core::QualityObject> mQualityObject;
+    o2::framework::Inputs mInputs;
+
+    // Depends - might be a (1) global checker output (more performant) or (2) per checkname output (more flexible)
```

Comment:
I would vote for separate output for each check, which might prove to be useful if we implelemnt the alarms mechanism inside DPL. @Barthelemy What do you think?